// dy-custom-display/esm2022/lib/service/dy-text-editor/dy-text-editor.service.mjs
import { Injectable } from '@angular/core';
import { BehaviorSubject, Subject } from "rxjs";
import * as i0 from "@angular/core";
export class DyTextEditorService {
    constructor() {
        this.displayEditorSubject = new BehaviorSubject(false);
        this.newEditorData = new Subject();
        this.editorDataChange = new Subject();
    }
    newChanges(key, value) {
        this.editorDataChange.next({ key: key, value: value });
    }
    addNewEditorData(key, value) {
        this.newEditorData.next({ key: key, value: value });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyTextEditorService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyTextEditorService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyTextEditorService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [] });
