// dy-custom-display/esm2022/lib/custom-display.module.mjs
import { NgModule } from '@angular/core';
import { DragDropModule } from "@angular/cdk/drag-drop";
import { MatButtonToggleModule } from "@angular/material/button-toggle";
import { MatButtonModule } from "@angular/material/button";
import { MatFormFieldModule } from "@angular/material/form-field";
import { MatSelectModule } from "@angular/material/select";
import { MatChipsModule } from "@angular/material/chips";
import { MatCheckboxModule } from "@angular/material/checkbox";
import { CommonModule } from "@angular/common";
import { DyDisplayDirective } from "./directives/dy-display.directive";
import { DyTranslateDirective } from "./directives/dy-translate.directive";
import { MatIcon } from "@angular/material/icon";
import * as i0 from "@angular/core";
export class CustomDisplayModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: CustomDisplayModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.10", ngImport: i0, type: CustomDisplayModule, declarations: [DyDisplayDirective], imports: [CommonModule,
            DragDropModule,
            MatButtonToggleModule,
            MatButtonModule,
            MatFormFieldModule,
            MatSelectModule,
            MatChipsModule,
            MatCheckboxModule,
            DyTranslateDirective,
            MatIcon], exports: [DyDisplayDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: CustomDisplayModule, imports: [CommonModule,
            DragDropModule,
            MatButtonToggleModule,
            MatButtonModule,
            MatFormFieldModule,
            MatSelectModule,
            MatChipsModule,
            MatCheckboxModule,
            MatIcon] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: CustomDisplayModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        DyDisplayDirective,
                    ],
                    imports: [
                        CommonModule,
                        DragDropModule,
                        MatButtonToggleModule,
                        MatButtonModule,
                        MatFormFieldModule,
                        MatSelectModule,
                        MatChipsModule,
                        MatCheckboxModule,
                        DyTranslateDirective,
                        MatIcon,
                    ],
                    exports: [
                        DyDisplayDirective,
                    ]
                }]
        }] });
