// dy-custom-display/esm2022/lib/service/notifications/notification.service.mjs
import { Injectable } from '@angular/core';
import { Subject } from "rxjs";
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/snack-bar";
export class NotificationService {
    constructor(_snackBar) {
        this._snackBar = _snackBar;
        this._successMessage = new Subject();
        this._errorMessage = new Subject();
    }
    get successMessage() {
        return this._successMessage;
    }
    get errorMessage() {
        return this._errorMessage;
    }
    newMessage(message, snack = false) {
        if (snack) {
            this._snackBar.open(message, 'Fermer', { duration: 5000 });
        }
        this._successMessage.next(message);
    }
    newError(message, snack = false) {
        if (snack) {
            this._snackBar.open(message, 'Fermer', {
                duration: 60 * 1000,
                panelClass: 'error'
            });
        }
        this._errorMessage.next(message);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: NotificationService, deps: [{ token: i1.MatSnackBar }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: NotificationService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: NotificationService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.MatSnackBar }] });
