// dy-custom-display/esm2022/lib/service/loading/loading.service.mjs
import { computed, Injectable, signal } from '@angular/core';
import * as i0 from "@angular/core";
export class LoadingService {
    static { this.numberOfLoadingProcess = signal(0); }
    constructor() {
    }
    static startLoading() {
        LoadingService.numberOfLoadingProcess.set(LoadingService.numberOfLoadingProcess() + 1);
    }
    static stopLoading() {
        LoadingService.numberOfLoadingProcess.set(LoadingService.numberOfLoadingProcess() - 1);
        if (LoadingService.numberOfLoadingProcess() < 0) {
            console.error("LoadingService: negative number of loading processes");
        }
    }
    static isLoading() {
        return computed(() => { return LoadingService.numberOfLoadingProcess() != 0; });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: LoadingService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: LoadingService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: LoadingService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [] });
