// dy-custom-display/esm2022/lib/service/user/user.service.mjs
import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
import { map } from "rxjs";
import { isPlatformBrowser } from "@angular/common";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
export class UserService {
    constructor(http, backendUrl, _platformId) {
        this.http = http;
        this.backendUrl = backendUrl;
        this._platformId = _platformId;
    }
    getUser() {
        return this.http.get(this.backendUrl + '/user', { transferCache: false }).pipe(map(x => {
            this.user = x;
            return x;
        }));
    }
    isAdmin() {
        return this.user ? this.hasAdminRole(this.user.roles) : this.backendUrl.includes('localhost');
    }
    isSuperAdmin() {
        return (this.user && this.user.email) ? (this.hasAdminRole(this.user.roles) && this.user.email.includes('dahn')) : false;
    }
    hasAdminRole(roles) {
        return roles.includes('ROLE_ADMIN') || this.backendUrl.includes('localhost');
    }
    isAdminOrRedirect() {
        if (isPlatformBrowser(this._platformId)) {
            this.getUser().subscribe({
                next: user => {
                    if (!this.hasAdminRole(user.roles)) {
                        window.location.href = this.backendUrl + "/login";
                    }
                },
                error: err => {
                    console.log(err);
                    if (!this.backendUrl.includes('localhost')) {
                        window.location.href = this.backendUrl + "/login";
                    }
                }
            });
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: UserService, deps: [{ token: i1.HttpClient }, { token: 'backendUrl' }, { token: PLATFORM_ID }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: UserService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: UserService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Inject,
                    args: ['backendUrl']
                }] }, { type: Object, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }] });
