// dy-custom-display/esm2022/lib/components/images/add-dyimage/add-dyimage.component.mjs
import { Component, EventEmitter, Output } from '@angular/core';
import { ReactiveFormsModule } from "@angular/forms";
import { MatDialogModule } from "@angular/material/dialog";
import { MatButtonModule } from "@angular/material/button";
import { MatIconModule } from "@angular/material/icon";
import * as i0 from "@angular/core";
import * as i1 from "../../../service/image-upload/image-upload.service";
import * as i2 from "@angular/material/button";
import * as i3 from "@angular/material/icon";
export class AddDyimageComponent {
    constructor(service) {
        this.service = service;
        this.image = new EventEmitter();
    }
    openDialog() {
        this.service.openDialog().subscribe({
            next: output => {
                this.image.emit(output);
            },
            error: (err) => {
                console.log('error with uploading image');
                console.log(err);
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: AddDyimageComponent, deps: [{ token: i1.ImageUploadService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.10", type: AddDyimageComponent, isStandalone: true, selector: "lib-add-dyimage", outputs: { image: "image" }, ngImport: i0, template: "<div>\n  <button (click)=\"openDialog()\" mat-raised-button>\n    <mat-icon fontIcon=\"add_circle\"></mat-icon>\n    Rajouter une nouvelle image\n  </button>\n</div>\n", styles: [""], dependencies: [{ kind: "ngmodule", type: ReactiveFormsModule }, { kind: "ngmodule", type: MatButtonModule }, { kind: "component", type: i2.MatButton, selector: "    button[mat-button], button[mat-raised-button], button[mat-flat-button],    button[mat-stroked-button]  ", exportAs: ["matButton"] }, { kind: "ngmodule", type: MatDialogModule }, { kind: "ngmodule", type: MatIconModule }, { kind: "component", type: i3.MatIcon, selector: "mat-icon", inputs: ["color", "inline", "svgIcon", "fontSet", "fontIcon"], exportAs: ["matIcon"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: AddDyimageComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lib-add-dyimage', standalone: true, imports: [
                        ReactiveFormsModule,
                        MatButtonModule, MatDialogModule, MatIconModule,
                    ], template: "<div>\n  <button (click)=\"openDialog()\" mat-raised-button>\n    <mat-icon fontIcon=\"add_circle\"></mat-icon>\n    Rajouter une nouvelle image\n  </button>\n</div>\n" }]
        }], ctorParameters: () => [{ type: i1.ImageUploadService }], propDecorators: { image: [{
                type: Output
            }] } });
