// dy-custom-display/esm2022/lib/lib/title/my-title-strategy.mjs
import { Inject, Injectable } from "@angular/core";
import { TitleStrategy } from "@angular/router";
import * as i0 from "@angular/core";
import * as i1 from "@angular/platform-browser";
import * as i2 from "@ngx-translate/core";
export class MyTitleStrategy extends TitleStrategy {
    constructor(title, translate, siteTitleBase) {
        super();
        this.title = title;
        this.translate = translate;
        this.siteTitleBase = siteTitleBase;
    }
    updateTitle(routerState) {
        const title = this.buildTitle(routerState);
        if (title !== undefined) {
            if (title.length > 0) {
                this.translate.stream(title).subscribe(res => {
                    this.title.setTitle(`${res} - ${this.siteTitleBase}`);
                });
            }
            else {
                this.title.setTitle(this.siteTitleBase);
            }
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: MyTitleStrategy, deps: [{ token: i1.Title }, { token: i2.TranslateService }, { token: 'siteTitleBase' }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: MyTitleStrategy }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: MyTitleStrategy, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: i1.Title }, { type: i2.TranslateService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: ['siteTitleBase']
                }] }] });
