// dy-custom-display/esm2022/lib/components/images/image-edit/image-edit.component.mjs
import { Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "@ngx-translate/core";
import * as i2 from "../../../service/translate/translation-client.service";
export class ImageEditComponent {
    constructor(service, clientService) {
        this.service = service;
        this.clientService = clientService;
        this.defaultUrl = '';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: ImageEditComponent, deps: [{ token: i1.TranslateService }, { token: i2.TranslationClientService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.10", type: ImageEditComponent, isStandalone: true, selector: "lib-image-edit", inputs: { defaultUrl: "defaultUrl", name: "name" }, ngImport: i0, template: "<p>image-edit works!</p>\n", styles: [""], dependencies: [{ kind: "ngmodule", type: CommonModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: ImageEditComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lib-image-edit', standalone: true, imports: [CommonModule], template: "<p>image-edit works!</p>\n" }]
        }], ctorParameters: () => [{ type: i1.TranslateService }, { type: i2.TranslationClientService }], propDecorators: { defaultUrl: [{
                type: Input
            }], name: [{
                type: Input,
                args: [{ required: true }]
            }] } });
