// dy-custom-display/esm2022/lib/service/swipe/swipe-manager.service.mjs
import { EventEmitter, Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class SwipeManagerService {
    constructor() {
        this.swipeRight = new EventEmitter();
        this.swipeLeft = new EventEmitter();
        this.xDown = null;
        this.yDown = null;
        this.handleStart = (event) => {
            this.handleTouchStart(event);
        };
        this.handleMove = (event) => {
            this.handleTouchMove(event);
        };
        this.keyPress = (event) => {
            if (event.code == 'ArrowLeft') {
                this.left();
            }
            if (event.code == 'ArrowRight') {
                this.right();
            }
        };
    }
    getTouches(evt) {
        return evt.touches || // browser API
            evt.originalEvent.touches; // jQuery
    }
    handleTouchStart(evt) {
        const firstTouch = this.getTouches(evt)[0];
        this.xDown = firstTouch.clientX;
        this.yDown = firstTouch.clientY;
    }
    ;
    right() {
        console.log('right');
        this.swipeRight.emit();
    }
    left() {
        console.log('left');
        this.swipeLeft.emit();
    }
    handleTouchMove(evt) {
        if (!this.xDown || !this.yDown) {
            return;
        }
        const xUp = evt.touches[0].clientX;
        const yUp = evt.touches[0].clientY;
        const xDiff = this.xDown - xUp;
        const yDiff = this.yDown - yUp;
        if (Math.abs(xDiff) > Math.abs(yDiff)) { /*most significant*/
            if (xDiff > 0) {
                /* right swipe */
                this.right();
            }
            else {
                /* left swipe */
                this.left();
            }
        }
        else {
            if (yDiff > 0) {
                /* down swipe */
            }
            else {
                /* up swipe */
            }
        }
        /* reset values */
        this.xDown = null;
        this.yDown = null;
    }
    ;
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: SwipeManagerService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: SwipeManagerService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: SwipeManagerService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [] });
