// dy-custom-display/esm2022/lib/service/title/title.service.mjs
import { Inject, Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/platform-browser";
export class TitleService {
    constructor(title, siteTitleBase) {
        this.title = title;
        this.siteTitleBase = siteTitleBase;
    }
    setTitle(newTitle) {
        this.title.setTitle(newTitle);
    }
    appendToTitle(newTitle) {
        this.setTitle(this.title.getTitle() + ' - ' + newTitle);
    }
    appendToBaseTitle(newTitle) {
        this.setTitle(this.siteTitleBase + ' - ' + newTitle);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: TitleService, deps: [{ token: i1.Title }, { token: 'siteTitleBase' }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: TitleService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: TitleService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.Title }, { type: undefined, decorators: [{
                    type: Inject,
                    args: ['siteTitleBase']
                }] }] });
