// dy-custom-display/esm2022/lib/components/notification/notification.component.mjs
import { Component } from '@angular/core';
import { NgIf } from "@angular/common";
import { DyTranslateDirective } from "../../directives/dy-translate.directive";
import * as i0 from "@angular/core";
import * as i1 from "../../service/notifications/notification.service";
export class NotificationComponent {
    constructor(service) {
        this.service = service;
        this.show = false;
        this.error = false;
        this.message = '';
        this.timeout = -1;
        this.service.successMessage.subscribe((msg) => {
            this.message = msg;
            this.show = true;
            this.error = false;
            this.prepareStop();
        });
        this.service.errorMessage.subscribe((msg) => {
            this.message = msg;
            this.show = true;
            this.error = true;
            this.prepareStop();
        });
    }
    prepareStop() {
        clearTimeout(this.timeout);
        this.timeout = setTimeout(() => {
            console.log('end');
            this.show = false;
        }, 8000);
    }
    stop() {
        this.show = false;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: NotificationComponent, deps: [{ token: i1.NotificationService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.10", type: NotificationComponent, isStandalone: true, selector: "app-notification", ngImport: i0, template: "<div *ngIf=\"show\" [class.show]=\"show\" aria-atomic=\"true\" aria-live=\"assertive\" class=\"toast\" role=\"alert\">\n  <div class=\"toast-header\">\n    <strong *ngIf=\"!error\" [appDyTranslate]=\"'notification.successHeader'\" class=\"me-auto\">Notification</strong>\n    <strong *ngIf=\"error\" [appDyTranslate]=\"'notification.errorHeader'\" class=\"me-auto\">Error</strong>\n    <button (click)=\"stop()\" aria-label=\"Close\" class=\"btn-close\" data-bs-dismiss=\"toast\" type=\"button\"></button>\n  </div>\n  <div class=\"toast-body\">\n    {{ message }}\n  </div>\n</div>\n", styles: [".toast{display:block;position:fixed;top:50%;left:50%;margin-left:-150px;margin-top:-50px;z-index:999}\n"], dependencies: [{ kind: "directive", type: NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: DyTranslateDirective, selector: "[appDyTranslate]", inputs: ["appDyTranslate"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: NotificationComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-notification', standalone: true, imports: [
                        NgIf,
                        DyTranslateDirective
                    ], template: "<div *ngIf=\"show\" [class.show]=\"show\" aria-atomic=\"true\" aria-live=\"assertive\" class=\"toast\" role=\"alert\">\n  <div class=\"toast-header\">\n    <strong *ngIf=\"!error\" [appDyTranslate]=\"'notification.successHeader'\" class=\"me-auto\">Notification</strong>\n    <strong *ngIf=\"error\" [appDyTranslate]=\"'notification.errorHeader'\" class=\"me-auto\">Error</strong>\n    <button (click)=\"stop()\" aria-label=\"Close\" class=\"btn-close\" data-bs-dismiss=\"toast\" type=\"button\"></button>\n  </div>\n  <div class=\"toast-body\">\n    {{ message }}\n  </div>\n</div>\n", styles: [".toast{display:block;position:fixed;top:50%;left:50%;margin-left:-150px;margin-top:-50px;z-index:999}\n"] }]
        }], ctorParameters: () => [{ type: i1.NotificationService }] });
