// dy-custom-display/esm2022/lib/service/image-upload/image-upload.service.mjs
import { Inject, Injectable } from '@angular/core';
import { filter, map, switchMap } from "rxjs";
import { UploadImageModalComponent } from "../../components/images/add-dyimage/upload-image-modal/upload-image-modal.component";
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/dialog";
import * as i2 from "@angular/common/http";
export class ImageUploadService {
    constructor(dialog, http, imageUploadUrl) {
        this.dialog = dialog;
        this.http = http;
        this.imageUploadUrl = imageUploadUrl;
    }
    uploadImage(file) {
        const formData = new FormData();
        formData.append('file', file, file.name);
        return this.http.post(this.imageUploadUrl + '/uploads-v2/', formData);
    }
    openDialog() {
        const dialogRef = this.dialog.open(UploadImageModalComponent);
        return dialogRef.afterClosed().pipe(filter(result => result?.alt && result?.file), switchMap((result) => {
            return this.uploadImage(result.file).pipe(map(uploadResponse => { return { alt: result.alt, urls: uploadResponse }; }));
        }));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: ImageUploadService, deps: [{ token: i1.MatDialog }, { token: i2.HttpClient }, { token: 'imageupload_url' }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: ImageUploadService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: ImageUploadService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.MatDialog }, { type: i2.HttpClient }, { type: undefined, decorators: [{
                    type: Inject,
                    args: ['imageupload_url']
                }] }] });
