// dy-custom-display/esm2022/lib/service/dy-contact/dy-contact.service.mjs
import { Inject, Injectable } from '@angular/core';
import { finalize, throwError } from "rxjs";
import { FormControl, FormGroup, Validators } from "@angular/forms";
import { LoadingService } from '../loading/loading.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
export class DyContactService {
    constructor(http, backendUrl) {
        this.http = http;
        this.backendUrl = backendUrl;
        this.formData = new FormGroup({
            name: new FormControl('', Validators.required),
            lastName: new FormControl(''),
            email: new FormControl('', [Validators.required, Validators.email]),
            message: new FormControl('', Validators.required)
        });
        this.formDataLong = new FormGroup({
            name: new FormControl('', Validators.required),
            nom: new FormControl('', Validators.required),
            lastName: new FormControl(''),
            phone: new FormControl('', [Validators.required]),
            email: new FormControl('', [Validators.required, Validators.email]),
            message: new FormControl('', Validators.required)
        });
        this.formLoadedTimestamp = Date.now();
    }
    sendContactForm() {
        return this.send(this.formData);
    }
    sendContactFormLong() {
        return this.send(this.formDataLong);
    }
    send(form) {
        if (form.invalid || (Date.now() - this.formLoadedTimestamp) < 5000 || form.getRawValue().lastName) {
            return throwError(() => 'form Invalid');
        }
        LoadingService.startLoading();
        return this.http.get(this.backendUrl + '/contact', { params: form.getRawValue() }).pipe(finalize(() => {
            LoadingService.stopLoading();
        }));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyContactService, deps: [{ token: i1.HttpClient }, { token: 'backendUrl' }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyContactService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyContactService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.HttpClient }, { type: undefined, decorators: [{
                    type: Inject,
                    args: ['backendUrl']
                }] }] });
