// dy-custom-display/esm2022/lib/components/images/tag/dy-img/dy-img.component.mjs
import { Component, computed, input, model, ViewChild } from '@angular/core';
import { DyTransitionDirective } from "../../../../directives/dy-transition.directive";
import * as i0 from "@angular/core";
export class DyImgComponent {
    constructor() {
        this.imgClass = input('');
        this.imgStyle = input('');
        this.image = input.required();
        this.alt = input.required();
        this.transitionName = input('none');
        this.sizeRatio = input(100);
        this.srcSet = computed(() => {
            return this.image().compressedUrls.map((url) => `${url.url} ${url.width}w`).join(', ');
        });
        this.sizes = computed(() => {
            return this.image().compressedUrls.map((url) => `(max-width: ${((Number(url.width) * 100) / this.sizeRatio())}px) ` + url.width + "px").join(', ');
        });
        this.height = model();
        this.width = model();
    }
    ngAfterViewChecked() {
        this.updateParentAttributes();
    }
    updateParentAttributes() {
        const img = this.imageElement.nativeElement;
        const parentAnchor = img.parentElement;
        if (parentAnchor) {
            // parentAnchor.setAttribute('href', img.src);
            if (img.complete) {
                this.updateAttributes(parentAnchor, img);
            }
            else {
                img.onload = () => {
                    this.updateAttributes(parentAnchor, img);
                };
            }
        }
    }
    updateAttributes(parentAnchor, img) {
        const originalWidth = img.naturalWidth;
        const originalHeight = img.naturalHeight;
        const windowHeight = window.innerHeight * 0.9;
        this.height.set(img.naturalHeight);
        this.width.set(img.naturalWidth);
        //   // Update parent <a> attributes
        //   parentAnchor.setAttribute('data-pswp-width', Math.round(originalWidth / originalHeight * windowHeight).toString());
        //   parentAnchor.setAttribute('data-pswp-height', windowHeight.toString());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyImgComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.1.0", version: "17.3.10", type: DyImgComponent, isStandalone: true, selector: "dy-img", inputs: { imgClass: { classPropertyName: "imgClass", publicName: "imgClass", isSignal: true, isRequired: false, transformFunction: null }, imgStyle: { classPropertyName: "imgStyle", publicName: "imgStyle", isSignal: true, isRequired: false, transformFunction: null }, image: { classPropertyName: "image", publicName: "image", isSignal: true, isRequired: true, transformFunction: null }, alt: { classPropertyName: "alt", publicName: "alt", isSignal: true, isRequired: true, transformFunction: null }, transitionName: { classPropertyName: "transitionName", publicName: "transitionName", isSignal: true, isRequired: false, transformFunction: null }, sizeRatio: { classPropertyName: "sizeRatio", publicName: "sizeRatio", isSignal: true, isRequired: false, transformFunction: null }, height: { classPropertyName: "height", publicName: "height", isSignal: true, isRequired: false, transformFunction: null }, width: { classPropertyName: "width", publicName: "width", isSignal: true, isRequired: false, transformFunction: null } }, outputs: { height: "heightChange", width: "widthChange" }, viewQueries: [{ propertyName: "imageElement", first: true, predicate: ["imageElement"], descendants: true }], ngImport: i0, template: "<img\n  #imageElement\n  [alt]=\"alt()\"\n  [class]=\"imgClass()\"\n  [dyTransition]=\"transitionName()\"\n  [sizes]=\"'(max-width: 768px) 100vw, '+sizeRatio()+'vw'\"\n  [src]=\"image().thumbnailUrl\"\n  [srcset]=\"srcSet()\"\n  [style]=\"imgStyle()\"\n/>\n", styles: [""], dependencies: [{ kind: "directive", type: DyTransitionDirective, selector: "[dyTransition]", inputs: ["dyTransition"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyImgComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dy-img', standalone: true, imports: [
                        DyTransitionDirective
                    ], template: "<img\n  #imageElement\n  [alt]=\"alt()\"\n  [class]=\"imgClass()\"\n  [dyTransition]=\"transitionName()\"\n  [sizes]=\"'(max-width: 768px) 100vw, '+sizeRatio()+'vw'\"\n  [src]=\"image().thumbnailUrl\"\n  [srcset]=\"srcSet()\"\n  [style]=\"imgStyle()\"\n/>\n" }]
        }], propDecorators: { imageElement: [{
                type: ViewChild,
                args: ['imageElement']
            }] } });
