// dy-custom-display/esm2022/lib/service/local-storage/local-storage.service.mjs
import { Inject, Injectable, PLATFORM_ID, signal } from '@angular/core';
import { isPlatformBrowser } from "@angular/common";
import { TranslationClientService } from "../translate/translation-client.service";
import * as i0 from "@angular/core";
export class LocalStorageService {
    constructor(_platformId, keys = []) {
        this._platformId = _platformId;
        this.values = new Map([]);
        const localKeys = [TranslationClientService.COOKIE_NAME, ...keys];
        // init
        localKeys.forEach(key => {
            this.values.set(key, signal(null));
        });
        if (isPlatformBrowser(this._platformId)) {
            localKeys.forEach(key => {
                this.values.get(key)?.set(this.getFromStorage(key));
            });
            // Listen for changes to local storage
            window.addEventListener('storage', (event) => {
                if (event.storageArea === localStorage) {
                    console.log('new localstorage data');
                    localKeys.forEach(key => {
                        this.values.get(key)?.set(this.getFromStorage(key));
                    });
                }
            });
        }
    }
    getFromStorage(key) {
        if (isPlatformBrowser(this._platformId)) {
            const item = localStorage.getItem(key);
            if (item) {
                try {
                    const { value, expiry } = JSON.parse(item);
                    if (!expiry || new Date(expiry) > new Date()) {
                        return value;
                    }
                    else {
                        localStorage.removeItem(key); // Remove expired token
                    }
                }
                catch (e) {
                    console.error('Error parsing JSON from localStorage:', e);
                    localStorage.removeItem(key); // Remove corrupted item
                }
            }
        }
        return null;
    }
    setValue(key, value, expiresInMs = 1000 * 60 * 60 * 24 * 21) {
        if (isPlatformBrowser(this._platformId)) {
            const signalValue = this.values.get(key);
            if (value && value != signalValue()) {
                const expiry = new Date().getTime() + expiresInMs;
                localStorage.setItem(key, JSON.stringify({ value, expiry }));
                signalValue.set(value);
            }
            else if (!value) {
                localStorage.removeItem(key);
                signalValue.set(null);
            }
        }
    }
    getValue(key) {
        return this.values.get(key).asReadonly();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: LocalStorageService, deps: [{ token: PLATFORM_ID }, { token: 'LOCAL_STORAGE_KEYS' }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: LocalStorageService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: LocalStorageService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: Object, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: ['LOCAL_STORAGE_KEYS']
                }] }] });
