// dy-custom-display/esm2022/lib/components/calendly/calendly.component.mjs
import { Component, Input } from '@angular/core';
import { CalendlyService } from "../../service/calendly-service/calendly.service";
import * as i0 from "@angular/core";
export class CalendlyComponent {
    ngOnInit() {
        CalendlyService.showEventEmitter.subscribe(x => {
            if (x) {
                this.show();
            }
        });
    }
    show() {
        // @ts-ignore
        Calendly.initPopupWidget({
            url: this.url
        });
        /*
        Calendly.initInlineWidget({
          url: this.url,
          parentElement: document.getElementById('calendly-embed')
        });*/
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: CalendlyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.10", type: CalendlyComponent, isStandalone: true, selector: "lib-calendly", inputs: { url: "url" }, ngImport: i0, template: "<link href=\"https://assets.calendly.com/assets/external/widget.css\" rel=\"stylesheet\">", styles: ["::ng-deep body .calendly-overlay .calendly-popup{max-height:1000px!important}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: CalendlyComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lib-calendly', standalone: true, imports: [], template: "<link href=\"https://assets.calendly.com/assets/external/widget.css\" rel=\"stylesheet\">", styles: ["::ng-deep body .calendly-overlay .calendly-popup{max-height:1000px!important}\n"] }]
        }], propDecorators: { url: [{
                type: Input,
                args: [{ required: true }]
            }] } });
