// dy-custom-display/esm2022/lib/directives/dy-display.directive.mjs
import { Directive, Input } from '@angular/core';
import { combineLatest, Subject, takeUntil } from "rxjs";
import * as i0 from "@angular/core";
import * as i1 from "@ngx-translate/core";
import * as i2 from "../service/translate/translation-client.service";
export class DyDisplayDirective {
    constructor(el, service, clientService) {
        this.el = el;
        this.service = service;
        this.clientService = clientService;
        this.destroy$ = new Subject();
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    ngAfterViewInit() {
        combineLatest([this.service.get(this.appDyDisplay), this.clientService.editSubject])
            .pipe(takeUntil(this.destroy$))
            .subscribe(([translateValue, editValue]) => {
            // Do something with both values
            if (translateValue != this.appDyDisplay || editValue) {
                this.el.nativeElement.style.display = 'block';
            }
            else {
                this.el.nativeElement.style.display = 'none';
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyDisplayDirective, deps: [{ token: i0.ElementRef }, { token: i1.TranslateService }, { token: i2.TranslationClientService }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "17.3.10", type: DyDisplayDirective, selector: "[appDyDisplay]", inputs: { appDyDisplay: "appDyDisplay" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyDisplayDirective, decorators: [{
            type: Directive,
            args: [{
                    standalone: false,
                    selector: '[appDyDisplay]'
                }]
        }], ctorParameters: () => [{ type: i0.ElementRef }, { type: i1.TranslateService }, { type: i2.TranslationClientService }], propDecorators: { appDyDisplay: [{
                type: Input,
                args: [{ required: true }]
            }] } });
