// dy-custom-display/esm2022/lib/directives/dy-transition.directive.mjs
import { Directive, HostListener, Inject, input, PLATFORM_ID } from '@angular/core';
import { Subscription } from "rxjs";
import { isPlatformBrowser } from "@angular/common";
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class DyTransitionDirective {
    constructor(elementRef, router, _platformId) {
        this.elementRef = elementRef;
        this.router = router;
        this._platformId = _platformId;
        this.subscription = new Subscription();
        this.dyTransition = input.required();
    }
    ngOnDestroy() {
        if (this.intervalId) {
            clearInterval(this.intervalId);
        }
        this.subscription.unsubscribe();
    }
    ngOnInit() {
        if (this.dyTransition() != 'none') {
            this.elementRef.nativeElement.style['view-transition-name'] = this.dyTransition();
        }
        // Listen to router events
        this.subscription.add(this.router.events.subscribe(event => {
            if (event) {
                this.checkVisibility();
            }
        }));
    }
    isElementInViewport(el) {
        const rect = el.getBoundingClientRect();
        const tolerance = 0.1; // 10% tolerance
        const windowHeight = window.innerHeight || document.documentElement.clientHeight;
        const windowWidth = window.innerWidth || document.documentElement.clientWidth;
        const topTolerance = tolerance * windowHeight;
        const leftTolerance = tolerance * windowWidth;
        return (rect.top >= -topTolerance &&
            rect.left >= -leftTolerance &&
            rect.bottom <= (1 + tolerance) * windowHeight &&
            rect.right <= (1 + tolerance) * windowWidth);
    }
    checkVisibility() {
        if (isPlatformBrowser(this._platformId)) {
            if (this.dyTransition() != 'none') {
                if (this.isElementInViewport(this.elementRef.nativeElement)) {
                    this.elementRef.nativeElement.style['view-transition-name'] = this.dyTransition();
                }
                else {
                    setTimeout(() => {
                        this.elementRef.nativeElement.style['view-transition-name'] = 'none';
                    }, 1);
                }
            }
        }
        else {
            if (this.dyTransition() != 'none') {
                this.elementRef.nativeElement.style['view-transition-name'] = this.dyTransition();
            }
        }
    }
    onScroll() {
        this.checkVisibility();
    }
    ngAfterViewInit() {
        this.checkVisibility(); // Initial check
        if (isPlatformBrowser(this._platformId)) {
            this.intervalId = setInterval(() => {
                this.checkVisibility();
            }, 500); // 500 milliseconds interval
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyTransitionDirective, deps: [{ token: i0.ElementRef }, { token: i1.Router }, { token: PLATFORM_ID }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "17.1.0", version: "17.3.10", type: DyTransitionDirective, isStandalone: true, selector: "[dyTransition]", inputs: { dyTransition: { classPropertyName: "dyTransition", publicName: "dyTransition", isSignal: true, isRequired: true, transformFunction: null } }, host: { listeners: { "window:scroll": "onScroll($event)", "window:click": "onScroll($event)" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.10", ngImport: i0, type: DyTransitionDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[dyTransition]',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: i0.ElementRef }, { type: i1.Router }, { type: Object, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }], propDecorators: { onScroll: [{
                type: HostListener,
                args: ['window:scroll', ['$event']]
            }, {
                type: HostListener,
                args: ['window:click', ['$event']]
            }] } });
